code/modules/mob/living/simple_animal/abnormality/waw/generalb.dm

```diff
@@ -0,0 +1,168 @@
+	SLEEP_CHECK_DEATH(80)
+	var/turf/T = pick(GLOB.department_centers)
+	forceMove(T)
+	icon_living = "general_breach"
+	icon_state = icon_living
+	var/turf/orgin = get_turf(src)
+	var/list/all_turfs = RANGE_TURFS(2, orgin)
+	for(var/turf/Y in all_turfs)
+		if(prob(60))
+			new /mob/living/simple_animal/hostile/worker_bee/soldier(Y)
+		else if(prob(20))
+			new /mob/living/simple_animal/hostile/artillery_bee(T)
+	..()
+
+/mob/living/simple_animal/hostile/abnormality/general_b/proc/spawn_bees()
+	var/X = pick(GLOB.xeno_spawn)
+	var/turf/T = get_turf(X)
+	new /mob/living/simple_animal/hostile/artillery_bee(T)
+	for(var/y = 1 to 5)
+		new /mob/living/simple_animal/hostile/worker_bee/soldier(T)
+
+/* Soldier bees */
+/mob/living/simple_animal/hostile/worker_bee/soldier
+	name = "soldier bee"
+	desc = "A disfigured creature with nasty fangs, and a snazzy cap"
+	icon_state = "soldier_bee"
+	icon_living = "soldier_bee"
+
+/* Artillery bees */
+/mob/living/simple_animal/hostile/artillery_bee
+	name = "artillery bee"
+	desc = "A disfigured creature with nasty fangs, and an oversized thorax"
+	icon = 'ModularTegustation/Teguicons/48x96.dmi'
+	icon_state = "artillerysergeant"
+	icon_living = "artillerysergeant"
+	friendly_verb_continuous = "scorns"
+	friendly_verb_simple = "scorns"
+	pixel_x = -8
+	base_pixel_x = -8
+	pixel_y = -8
+	base_pixel_y = -8
+
+	var/fire_cooldown_time = 10 SECONDS
+	var/fire_cooldown
+	var/fireball_range = 30
+
+/mob/living/simple_animal/hostile/artillery_bee/Initialize()
+	fire_cooldown = world.time + fire_cooldown_time
+	..()
+
+/mob/living/simple_animal/hostile/artillery_bee/Life()
+	. = ..()
+	if(!.) // Dead
+		return FALSE
+	if((fire_cooldown < world.time))
+		fireshell()
+
+/mob/living/simple_animal/hostile/artillery_bee/proc/fireshell()
+	fire_cooldown = world.time + fire_cooldown_time
+	var/list/targets = list()
+	for(var/mob/living/carbon/human/L in livinginrange(fireball_range, src))
+		if(faction_check_mob(L, FALSE))
+			continue
+		if(L.stat == DEAD)
+			continue
+		targets += L
+	new /obj/effect/beeshell(get_turf(pick(targets)))
+
+/obj/effect/beeshell
+	name = "bee shell"
+	desc = "A target warning you of incoming pain"
+	icon = 'icons/effects/effects.dmi'
+	icon_state = "beetillery"
+	move_force = INFINITY
+	pull_force = INFINITY
+	generic_canpass = FALSE
+	movement_type = PHASING | FLYING
+	var/boom_damage = 100 //Half Red, Half Black
+
+/obj/effect/beeshell/Initialize()
+	..()
+	addtimer(CALLBACK(src, .proc/explode), 2 SECONDS)
+
+//Smaller Scorched Girl bomb
+/obj/effect/beeshell/proc/explode()
+	playsound(get_turf(src), 'sound/abnormalities/scorchedgirl/explosion.ogg', 125, 0, 8)
+	for(var/mob/living/carbon/human/H in view(3, src))
+		H.apply_damage(boom_damage, RED_DAMAGE, null, H.run_armor_check(null, RED_DAMAGE), spread_damage = TRUE)
+		H.apply_damage(boom_damage, BLACK_DAMAGE, null, H.run_armor_check(null, BLACK_DAMAGE), spread_damage = TRUE)
```

Comment:
Why does it deal doule damage of both black and red..?

Reply:
It deals 100 black, 100 red. IDK. It's a bomb, and High red resist is everywhere.

Reply:
Could you also use some other sound instead of re-using scorched girl explosion?